src/lib.rs

```diff
@@ -261,7 +278,7 @@ impl<const OSC_RAW_BUF_SIZE: usize> Parser<OSC_RAW_BUF_SIZE> {
                 let idx = self.osc_raw.len();
 
                 // Param separator
-                if byte == b';' {
+                if self.str_start == 0x1d && byte == b';' {
```

Comment:
One extra branch isn't going to be the end of the world, but I don't see how this isn't just going to be a net-negative for anyone who doesn't use APC escapes (which is basically everyone).